torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -495,6 +600,14 @@ std::vector<std::vector<size_t>> compute_bucket_assignment_by_size(
   for (size_t i = 0; i < tensors.size(); i++) {
     const auto& tensor = tensors[i];
     AT_ASSERTM(!tensor.is_sparse(), "No support for sparse tensors.");
```

Comment:
Is this check still necessary? Or will it be possible that `is_sparse()` returns different values for tensor and its gradient?

Reply:
Yes, the parameter itself will be dense.